src/main/java/org/opensearch/ad/transport/ProfileTransportAction.java

```diff
@@ -45,28 +45,23 @@ public class ProfileTransportAction extends TransportAction<ProfileRequest, Prof
     private ModelManager modelManager;
     private FeatureManager featureManager;
     private CacheProvider cacheProvider;
-    private SDKClusterService clusterService;
+    private SDKClusterService sdkClusterService;
     // the number of models to return. Defaults to 10.
     private volatile int numModelsToReturn;
 
     /**
      * Constructor
      *
-     * @param threadPool ThreadPool to use
-     * @param clusterService ClusterService
-     * @param transportService TransportService
      * @param actionFilters Action Filters
      * @param modelManager model manager object
      * @param featureManager feature manager object
      * @param cacheProvider cache provider
-     * @param settings Node settings accessor
      */
     @Inject
     public ProfileTransportAction(
         ExtensionsRunner extensionsRunner,
         ActionFilters actionFilters,
         TaskManager taskManager,
-        SDKClusterService clusterService,
```

Comment:
the cluster service is already bound to guice, we can inject it directly here instead of retrieving it from the extensions runner

Reply:
I see, I thought the other way, as We are already passing extensions runner so we can utilize it rather than passing a whole new argument.

Reply:
I would prefer that we utilize guice where we can, @owaiskazi19 do you have any strong opinions about this?

Reply:
Agree with @joshpalis .... note we have to use the ExtensionsRunner in the Rest Actions and there it's appropriate to minimize parameters but here let's inject.